src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -289,48 +289,75 @@ private void VerifyDiagnosticResults(IEnumerable<Diagnostic> actualResults, Immu
 
                 if (!expected.HasLocation)
                 {
-                    verifier.Equal(Location.None, actual.Location, $"Expected:\r\nA project diagnostic with No location\r\nActual:\r\n{FormatDiagnostics(analyzers, actual)}");
+                    message = $"Expected a project diagnostic with no location:{Environment.NewLine}" +
```

Comment:
perhaps we can add some extensions to the `verifier` so we don't need to repeat all this message formatting code everywhere

Reply:
Extracted a helper method for this